qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -71,98 +103,110 @@ def _default_options(cls):
         return Options(shots=1024)
 
     @staticmethod
-    def _get_info(inst):
-        p_dict = inst.operation
-        qubit = tuple(int(str(val)[-2]) for val in inst.qubits)
+    def _get_info(instruction: CircuitInstruction) -> Tuple[Tuple(int), Tuple(float), str]:
+        p_dict = instruction.operation
+        qubit = tuple(int(str(val)[-2]) for val in instruction.qubits)
         params = tuple(float(val) for val in p_dict.params)
         return qubit, params, p_dict.name
 
     @staticmethod
     def _state_vector_to_result(
         state: Union[Statevector, np.ndarray],
-        shots: int,
-        meas_return: MeasReturnType,
-        meas_level: MeasLevel,
-    ):
+        shots: Optional[int] = 1024,
+        meas_return: Optional[MeasReturnType] = 0,
+        meas_level: Optional[MeasLevel] = 0,
+    ) -> Union[Dict[str, int], complex]:
         """Convert the state vector to IQ data or counts."""
 
-        # if meas_level == MeasLevel.CLASSIFIED:
-        # sample from the state vector. There might already be functions in Qiskit to do this.
-        #     counts = ...
-        # if meas_level == MeasLevel.KERNELED:
-        #     iq_data = ... create IQ data.
-        pass
-
-    def solve(self, schedule_blocks, qubits):
-        """Soleves a single schdule block and returns the unitary"""
+        if meas_level == MeasLevel.CLASSIFIED:
+            measurement = Statevector(state).sample_counts(shots)
+        elif meas_level == MeasLevel.KERNELED:
+            raise QiskitError("TODO: generate IQ data")
+            # measurement = iq_data = ... #create IQ data.
+            # Q5
```

Comment:
For this part we can get some inspiration with what is already done. Perhaps this backend should have some cluster centers (0, 1, and 2) with a variance and then we can produce IQ shots. See e.g. https://github.com/Qiskit/qiskit-experiments/blob/1137029588a4f3fc4a414e16077da7cb25ab426a/qiskit_experiments/test/mock_iq_backend.py#L308 Perhaps we can port a good part of this to this PR.